code.js

```diff
@@ -1,3 +1,64 @@
-function dijkstra(graph, sourceNode) {
-    return [];
+function dijkstra(graph) {
+    return dijkstraHelper(graph, 0);
+}
+
+function minDistance(dist,sptSet)
+{
+	let min = Infinity;
+	let min_index = -1;
+	
+	for(i = 0; i < dist.length; i++)
+	{
+		if (sptSet[i] == false && dist[i] <= min) 
+		{
+			min = dist[i];
+			min_index = i;
+		}
+	}
+	return min_index;
+}
+
+function printSolution(dist)
+{
+	console.log("Vertex \t\t Distance from Source");
+	for(let i = 0; i < dist.length; i++)
+	{
+		console.log(i + " \t\t " + 
+				dist[i]);
+	}
+}
+
+
+function dijkstraHelper(graph, src)
+{
+	let dist = new Array(graph.length);
+	let sptSet = new Array(graph.length);
+
+	for(let i = 0; i < graph.length; i++)
+	{
+		dist[i] = Infinity;
+		sptSet[i] = false;
+	}
+
+	dist[src] = 0;
+	
+	for(let count = 0; count < graph.length - 1; count++)
+	{
+		let u = minDistance(dist, sptSet);
```

Comment:
What's the complexity of `minDistance()` and how does that affect the overall complexity?

Reply:
minDistance() is reliant on dist.lenght, or otherwise known as V in regards to complexity. Overall this results in $\Theta(v^2 + v^2)$ which results in $\Theta(v^2 + v^2)